### ISSUE TITLE
SIGABRT error when using stories.mdx

### ISSUE THREAD

**Issue by DoneDeal0** [storybookjs/storybook#12562]
**Describe the bug**
Storybook crashes when I write `stories.mdx` files. The error returned in the terminal is: 
> FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory. (...) 
> Command failed with signal "SIGABRT"

**To Reproduce**
Create a `stories.mdx` file, launch storybook. I have installed `react-is`, `babel-loader`, and `@storybook/addon-docs` + all other relevant packages. 

**Expected behavior**
Storybook should work fine, and display the mdx story. All other `stories.tsx` files were working fine before using .mdx. 

**Code snippets**
Here is my Storybook main.js file:
```javascript
module.exports = {
  stories: ["../**/stories.@(js|mdx|tsx)"],
  addons: [
    {
      name: "@storybook/addon-essentials",
      options: { backgrounds: false, viewport: false },
    },
    "storybook-addons-abstract/register",
    {
      name: "@storybook/addon-docs",
      options: {
        configureJSX: true,
        babelOptions: {},
        sourceLoaderOptions: null,
      },
    },
  ],
};
```

Here is my story:
```javascript
import { Story, Meta, Canvas } from "@storybook/components";
import Text from ".";

<Meta title="MDX/Text" component={Text} />

# Text
<Canvas>
  <Story name="defaut">
    <Text text="hello" />
  </Story>
</Canvas>
```

**System:**

```
Environment Info:

  System:
    OS: macOS 10.15.6
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 10.19.0 - ~/.nvm/versions/node/v10.19.0/bin/node
    Yarn: 1.22.5 - ~/Desktop/frontflip/node_modules/.bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v10.19.0/bin/npm
  Browsers:
    Chrome: 85.0.4183.121
    Firefox: 79.0
    Safari: 13.1.2
  npmPackages:
    @storybook/addon-docs: ^6.0.21 => 6.0.21
    @storybook/addon-essentials: ^6.0.20 => 6.0.21
    @storybook/addon-toolbars: ^6.0.20 => 6.0.21
    @storybook/addons: ^6.0.20 => 6.0.21
    @storybook/react: ^6.0.20 => 6.0.21
```

**Comment by shilman** [comment]
Do you have a repro repo you can share?

**Comment by DoneDeal0** [comment]
Unfortunately no, it is a private project. However, I may provide more code snippets if necessary.

**Comment by DoneDeal0** [comment]
SOLVED: Using node 12.18.3 and launching the app with --no-dll fixed the issue.
